django/wagtail/research/recommendation-of-adoption.md

```diff
@@ -0,0 +1,42 @@
+
+## For further exploration
+
+Though we've already had success with Wagtail in client work, the following areas warrant further exploration and documentation:
+
+1. **Create a pattern for content import/export process and media storage**
+
+    We should settle on a pattern for importing and exporting CMS content—this is helpful during development, and for new deployments of a site. For the Lugar project we have a management command for importing and a Docker command for exporting in the README; see [here](https://github.com/datamade/committee-oversight#initial-cms-content). This was successful, but debugging file storage was burdensome. For future projects Wagtail media uploads could be stored in S3, as detailed in [this blog post](https://wagtail.io/blog/amazon-s3-for-media-files/).
+
+2. **Create more reusable class-based views in Wagtail**
+
+    In the Lugar project we successfully implemented a [DetailPage model](https://github.com/datamade/committee-oversight/blob/master/committeeoversightapp/models.py#L482) to replicate the functionality of a [DetailView](https://ccbv.co.uk/projects/Django/3.0/django.views.generic.detail/DetailView/). We were generally pleased with this approach, as it allowed us to create detail views that both auto-populated data and allowed users to edit fields. This pattern could be extended to other Django [class-based views](https://ccbv.co.uk/) such as ListView. In the Lugar project, page creation was handled manually by the client. For future iterations of detail pages with different client needs, we may want to set up a system to automatically create a page for each record.
+
+3. **Using Wagtail for data management**
+
+    For the two projects in which DataMade has so far implemented Wagtail, users have not been able to change data related to Django models not originally created in Wagtail. For example, for the Lugar project there is a separate custom interface for entering data about individual hearings. However, there is support for editing any model through Wagtail’s [ModelAdmin](https://docs.wagtail.io/en/v2.8/reference/contrib/modeladmin/) module. For simple data management needs this could be very powerful and warrants further investigation. It could alleviate our need for custom data management tool development on some projects, and would mean users only have to learn to use one interface. See [this how-to issue](https://github.com/datamade/how-to/issues/35).
+
+4. **Explore using Wagtail for non-Django sites**
+
+    As Wagtail is a headless CMS, it’s possible to use it with frameworks other than Django. For example, [here is an example of the Wagtail team building a site with Gatsby](https://wagtail.io/blog/using-gatsby-wagtail-build-case-study/). This has not yet been successfully implemented at DataMade, and could be a good R&D thread to pull on sometime.
```

Comment:
I started some work in this direction here: https://github.com/datamade/how-to/issues/31#issuecomment-522728121 The main difficulty I ran into was that there's no built-in way to expose the site build lifecycle to content editors in a headless Wagtail instance. I explored writing a custom integration but it's not a trivial amount of work. I don't think this history necessarily needs to be included in this documentation, I just wanted to make sure everyone was aware of it.